source/routing/raptor_api.cpp

```diff
@@ -994,11 +1017,11 @@ parse_datetimes(RAPTOR& raptor,
                                      pbnavitia::DATE_OUT_OF_BOUNDS,
                                      "date is not in data production period");
         }
-        datetimes.push_back(ptime);
+        datetimes.push_back(std::make_pair(ptime, datetime));
     }
     if(clockwise)
         std::sort(datetimes.begin(), datetimes.end(),
-                  [](bt::ptime dt1, bt::ptime dt2){return dt1 > dt2;});
+                  [](std::pair<bt::ptime, uint64_t> dt1, std::pair<bt::ptime, uint64_t> dt2){return dt1.first > dt2.first;});
```

Comment:
minor: `[](const std::pair<bt::ptime, uint64_t>& dt1, const std::pair<bt::ptime, uint64_t>& dt2)` (when can we have c++14? :p )

Reply:
:smiley: 
:white_check_mark: